src/sagas/backend.ts

```diff
@@ -275,6 +276,80 @@ function* backendSaga(): SagaIterator {
       handleResponseError(resp);
     }
   });
+
+  yield takeEvery(actionTypes.FETCH_NOTIFICATIONS, function*(action) {
+    const tokens = yield select((state: IState) => ({
+      accessToken: state.session.accessToken,
+      refreshToken: state.session.refreshToken
+    }));
+
+    if (!tokens.accessToken) {
+      return;
+    }
+
+    const resp: Response | null = yield request('notification', 'GET', {
+      accessToken: tokens.accessToken,
+      refreshToken: tokens.refreshToken,
+      shouldAutoLogout: false
+    });
+    if (resp && resp.ok) {
+      const newNotifications: AcademyNotification[] = ((yield resp.json()) as any[]).map(n => {
```

Comment:
if you use a separate function and bind the result to a variable, you can avoid `yield resp.json()) as any[]`